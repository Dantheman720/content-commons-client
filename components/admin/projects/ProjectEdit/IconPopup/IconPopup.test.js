import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import IconPopup from './IconPopup';

const Component = <IconPopup message="A test message" iconType="info circle" size="tiny" />;

const newProps = {
  message: 'An updated test message',
  iconType: 'undo',
  size: 'huge'
};

describe( '<IconPopup />', () => {
  it( 'renders without crashing', () => {
    const wrapper = shallow( Component );
    expect( wrapper.exists() ).toEqual( true );
    expect( toJSON( wrapper ) ).toMatchSnapshot();
  } );

  it( 'receives props', () => {
    const wrapper = shallow( Component );
    expect( wrapper.prop( 'content' ) ).toEqual( 'A test message' );
    expect( wrapper.prop( 'size' ) ).toEqual( 'tiny' );
  } );

  it( 'sets new props', () => {
    const wrapper = shallow( Component );
    const { message, iconType, size } = newProps;
    wrapper.setProps( { message, iconType, size } );
    expect( wrapper.prop( 'content' ) ).toEqual( message );
    expect( wrapper.prop( 'size' ) ).toEqual( size );
  } );
} );
