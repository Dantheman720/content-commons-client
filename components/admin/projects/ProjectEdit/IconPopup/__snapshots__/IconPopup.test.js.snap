// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconPopup /> renders without crashing 1`] = `
<Popup
  content="A test message"
  inverted={true}
  keepInViewPort={true}
  on={
    Array [
      "hover",
      "click",
      "focus",
    ]
  }
  position="top left"
  size="tiny"
  trigger={
    <span>
      <Focusable
        order={0}
      >
        <Icon
          as="i"
          name="info circle"
          size="tiny"
        />
      </Focusable>
    </span>
  }
/>
`;
